entity_event_generator.tf

```diff
@@ -1,24 +1,60 @@
-  entity_event_topic_name  = "${local.environment}-entity-event-generator-topic"
-  entity_event_queue_name  = "${local.environment}-entity-event-queue"
-  last_polled_table_name   = "${local.environment}-entity-event-lambda-updated-since-query-start-datetime"
-  entity_event_topic_arn   = "arn:aws:sns:eu-west-2:${data.aws_caller_identity.current.account_id}:${local.entity_event_topic_name}"
+  entity_event_lambda_name_old = "${local.environment}-entity-event-generator-lambda"
+  entity_event_lambda_name     = "${local.environment}-dr2-entity-event-generator-lambda"
+  entity_event_topic_name_old  = "${local.environment}-entity-event-generator-topic"
+  entity_event_topic_name      = "${local.environment}-dr2-entity-event-generator-topic"
+  entity_event_queue_name_old  = "${local.environment}-entity-event-queue"
+  entity_event_queue_name      = "${local.environment}-dr2-entity-event-queue"
+  last_polled_table_name_old   = "${local.environment}-entity-event-lambda-updated-since-query-start-datetime"
+  last_polled_table_name       = "${local.environment}-dr2-entity-event-lambda-updated-since-query-start-datetime"
+  entity_event_topic_arn       = "arn:aws:sns:eu-west-2:${data.aws_caller_identity.current.account_id}:${local.entity_event_topic_name_old}"
 }
 
 module "entity_event_cloudwatch_event" {
   source                  = "git::https://github.com/nationalarchives/da-terraform-modules//cloudwatch_events"
   rule_name               = "${local.environment}-entity-event-schedule"
   schedule                = "rate(1 minute)"
-  lambda_event_target_arn = "arn:aws:lambda:eu-west-2:${data.aws_caller_identity.current.account_id}:function:${local.entity_event_lambda_name}"
+  lambda_event_target_arn = "arn:aws:lambda:eu-west-2:${data.aws_caller_identity.current.account_id}:function:${local.entity_event_lambda_name_old}"
 }
 
 module "entity_event_generator_lambda" {
   source        = "git::https://github.com/nationalarchives/da-terraform-modules//lambda"
-  function_name = local.entity_event_lambda_name
+  function_name = local.entity_event_lambda_name_old
   handler       = "uk.gov.nationalarchives.Lambda::handleRequest"
   policies = {
     entity_event_policy = templatefile("${path.module}/templates/iam_policy/entity_event_lambda_policy.json.tpl", {
+      account_id                 = data.aws_caller_identity.current.account_id
+      lambda_name                = local.entity_event_lambda_name_old
+      dynamo_db_arn              = module.entity_event_lambda_updated_since_query_start_datetime_table.table_arn
+      secrets_manager_secret_arn = aws_secretsmanager_secret.preservica_secret.arn
+      sns_arn                    = local.entity_event_topic_arn
+    })
+  }
+  timeout_seconds = 180
+  memory_size     = local.java_lambda_memory_size
+  runtime         = local.java_runtime
+  tags            = {}
+  lambda_invoke_permissions = {
+    "events.amazonaws.com" = module.entity_event_cloudwatch_event.event_arn
```

Comment:
Should this have `dr2` in front of it?

Reply:
This one shouldn't but the one in the `dr2_entity_event_generator_lambda` should. I've changed it. There were a few bits missing in this file which I've changed now.